packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -54,7 +54,8 @@ export async function createNewFluidFile(
     }
 
     const filePath = newFileInfo.filePath ? encodeURIComponent(`/${newFileInfo.filePath}`) : "";
-    const encodedFilename = encodeURIComponent(`${newFileInfo.filename}.fluid`);
+    const fileName = newFileInfo.filename.indexOf(".") === -1 ? `${newFileInfo.filename}.fluid` : newFileInfo.filename;
```

Comment:
I think filenames with "." are allowed, so need to think of  something else to check if file name has extension in it or not. Maybe we need to take it via Header in create new request path.